pkg/operator/ceph/cluster/controller.go

```diff
@@ -89,13 +90,32 @@ var ClusterResourceRookLegacy = opkit.CustomResource{
 	Kind:    reflect.TypeOf(cephbeta.Cluster{}).Name(),
 }
 
+// updaterCtx is the update handler context
+type updaterCtx struct {
+	context.Context
+	cancel context.CancelFunc
+	wait   chan struct{}
+}
+
+func newUpdaterCtx() *updaterCtx {
+	ctx, cancel := context.WithCancel(context.Background())
+	wait := make(chan struct{})
+
+	return &updaterCtx{
+		ctx,
+		cancel,
+		wait,
+	}
+}
+
 // ClusterController controls an instance of a Rook cluster
 type ClusterController struct {
 	context          *clusterd.Context
 	volumeAttachment attachment.Attachment
 	devicesInUse     bool
 	rookImage        string
 	clusterMap       map[string]*cluster
+	updaterCtxMap    map[string]*updaterCtx
```

Comment:
Instead of creating a new map, could we add the context to the `cluster` type?

Reply:
Yep. It will help share code between OnAdd and OnUpdate.